provider/src/main/java/com/cloudera/director/google/sql/GoogleCloudSQLInstanceTemplateConfigurationValidator.java

```diff
@@ -0,0 +1,131 @@
+  @VisibleForTesting
+  static final String INVALID_PREFIX_MSG = "Instance name prefix must follow this pattern: " +
+      "The first character must be a lowercase letter, and all following characters must be a dash, lowercase " +
+      "letter, or digit.";
+
+  /**
+   * The Google Cloud SQL provider.
+   */
+  private final GoogleCloudSQLProvider provider;
+
+  /**
+   * The pattern to which instance name prefixes must conform. The pattern is the same as that for instance names in
+   * general, except that we allow a trailing dash to be used. This is allowed since we always append a dash and the
+   * specified instance id to the prefix.
+   *
+   *  @see <a href="https://developers.google.com/resources/api-libraries/documentation/sqladmin/v1beta4/java/latest/com/google/api/services/sqladmin/model/DatabaseInstance.html#setName(java.lang.String)" />
+   */
+  private final static Pattern instanceNamePrefixPattern = Pattern.compile("[a-z][-a-z0-9]*");
+
+  /**
+   * Creates a Google Cloud SQL instance template configuration validator with the specified parameters.
+   *
+   * @param provider the Google Cloud SQL provider
+   */
+  public GoogleCloudSQLInstanceTemplateConfigurationValidator(GoogleCloudSQLProvider provider) {
+    this.provider = Preconditions.checkNotNull(provider, "provider");
+  }
+
+  @Override
+  public void validate(String name, Configured configuration,
+      PluginExceptionConditionAccumulator accumulator, LocalizationContext localizationContext) {
+
+    checkTier(configuration, accumulator, localizationContext);
+    checkPrefix(configuration, accumulator, localizationContext);
+  }
+
+  /**
+   * Validates the configured tier type.
+   *
+   * @param configuration       the configuration to be validated
+   * @param accumulator         the exception condition accumulator
+   * @param localizationContext the localization context
+   */
+  void checkTier(Configured configuration,
+      PluginExceptionConditionAccumulator accumulator,
+      LocalizationContext localizationContext) {
+
+    String tier = configuration.getConfigurationValue(TIER, localizationContext);
+
+    // TODO(kl3n1nz) Might want to make an API call instead.
+    // SQL API currently is supporting only list() method.
+    if (tier != null && !TIERS.contains(tier)) {
+      addError(accumulator, TIER, localizationContext, null, INVALID_TIER_MSG, tier);
+    }
+  }
+
+  /**
+   * Validates the configured prefix.
+   *
+   * @param configuration       the configuration to be validated
+   * @param accumulator         the exception condition accumulator
+   * @param localizationContext the localization context
+   */
+  static void checkPrefix(Configured configuration,
+      PluginExceptionConditionAccumulator accumulator,
+      LocalizationContext localizationContext) {
+
+    String instanceNamePrefix = configuration.getConfigurationValue(INSTANCE_NAME_PREFIX, localizationContext);
+
+    LOG.info(">> Validating prefix '{}'", instanceNamePrefix);
```

Comment:
It is for indentation. Cloudera uses it on AWS provider and we want to make the code consistent.